ci: automated releases with semantic release

privatenumber opened the pull request:
<!--
Prepend your PR title with one of the following:
- `feat:` A new feature
- `fix:` A bug fix
- `docs:` Documentation only changes
- `style:` Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- `refactor:` A code change that neither fixes a bug or adds a feature
- `perf:` A code change that improves performance
- `test:` Add missing tests
- `chore:` Changes to the build process or auxiliary tools and libraries such as documentation generation
-->

**Describe the problem prior to this PR (link ticket/issue):**
- Releasing a new version was a manual process.

**Describe the changes in this PR:**
<!-- Tip: inline screenshots to better communicate the changes -->
- [x] Added a release workflow that uses [semantic-release](https://github.com/semantic-release/semantic-release) to automatically determine the next version to bump to, and publish to both npm and GitHub.
- [x] Added more "special branches" that semantic-release will trigger pre-releases from
- [x] Cleaned up some workflow code
- [x] Add protected branches: `next`, `next-major`, `beta`, `alpha`
- [x] Dry-run instructions for local release testing


**Other information:**

@pretzelhammer LMK what you think and I can start testing it out.

pretzelhammer commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,18 +1,40 @@
-<!-- Tip: inline screenshots to better communicate the changes -->
+## Describe the changes in this PR
+<!--
+📸 Inline screenshots to better communicate the changes
```
Not sure if this is really needed since styleguides get auto-deployed for every PR, but I guess they could still copy&paste the link from the bot comment into their PR description.

privatenumber replied:
I think eventually the auto-deploys should expire/be cleaned.

Also something to think about if the assets are being pushed to the same repo is that it weights down the entire repo history. Each deploy will get slower and slower.

pretzelhammer replied:
ah yeah, I made an issue for that https://github.com/square/maker/issues/45

Also, I believe the history depth can be controlled when checking out a git branch/repo, I can't remember if I made that optimization in the workflow already but it should be possible to add so that the workflow doesn't get progressively slower when more commits are added to the `deploys` branch.

pretzelhammer commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,29 @@
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Use Node.js v14
```
Currently the `.nvmrc` for the repo is on v12.8.2 so can we use node 12.x instead? https://github.com/square/maker/blob/master/.nvmrc

Actually, what would probably be best is if we could have the job run `nvm use` so we don't have to update it when we update `.nvmrc`.

pretzelhammer commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,29 @@
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+      run: npx semantic-release
```
I'm assuming this command does-it-all, i.e.
- makes & commits git tag (based on package.json version)
- makes github release with commit log since last release
- publishes NPM package

privatenumber replied:
Yep!

pretzelhammer commented on .github/CONTRIBUTING.md:
````diff
@@ -126,53 +129,67 @@ If you want to test a branch before releasing it, you can push up a _built branc
-chore: enabled NUXT SSR
-```
+### Branches
+#### develop
````
If possible, I'd like to get rid of this additional branch dependency as part of Maker's regular git workflow. I think people should just make their PRs and merge them against `master`, I don't see what adding a `develop` branch gives us.

pretzelhammer commented on .github/CONTRIBUTING.md:
```diff
@@ -126,53 +129,67 @@ If you want to test a branch before releasing it, you can push up a _built branc
+#### master
+The latest stable version of the library. 
+
+PRs against `master` must only be from the `develop` branch for releasing purposes and merged via **merge commit**.
```
These are the kinds of details that trip people up all the time, merge commits from develop->master but squash commits from branches->develop is confusing and error prone. Can we just make it so that everyone just squash commits their PRs into `master`, which is how it currently works?

pretzelhammer commented on .github/CONTRIBUTING.md:
```diff
@@ -126,53 +129,67 @@ If you want to test a branch before releasing it, you can push up a _built branc
-The only "special" branch is the `master` branch which represents the latest stable version of the library. Most PRs with bug fixes, features, and even breaking changes should probably be made against, and ultimately merged into, `master`. The only exception is if you're unsure of changes and need to validate your ideas in production before making an official stable Maker release, which would be  ...
+This repository uses [Conventional Commits](https://www.conventionalcommits.org) for simple yet meaningful commit messages. Not only are they user-friendly, they are also machine-readable for automated release notes and versioning.
 
-Let's say Maker is on version `1.5.2` and you have a breaking change, or maybe several, that you'd like to make and that would require a major version release of `2.0.0`. Unless you're very confident in your changes you'll likely want to do a few _pre-releases_ first, and hold out on directly merging your changes directly into `master`. So what you'd do is:
```
I don't mind you removing these instructions but I think it's important we offer some guidance on how to do pre-releases. It's kind of uncommon to do pre-releases so I'd like an easy-to-follow step-by-step guide in the docs telling people how to do it.

privatenumber replied:
I added a reference to the [semantic-release docs](https://github.com/semantic-release/semantic-release/blob/master/docs/recipes/pre-releases.md#publishing-pre-releases) for this as they explain it very elaborately.

pretzelhammer commented on .github/dep-version-check/index.js:
```diff
@@ -1,67 +1,29 @@
-
-/* checkout master */
-
-(async function checkoutMasterAndPull() {
```
Ah, you got rid of this because the `semantic-release` script will generate the new version number automatically. Okay.

I think we should document this. It's not immediately obvious to everyone that if they write a `breaking: something` commit that it will make a major version release of the library, or that perhaps `chore: etc` will not bump the version number at all.

privatenumber replied:
I think I addressed this in the PR template and Contribution docs now, but LMK if there's more I can do.

pretzelhammer commented on .github/CONTRIBUTING.md:
````diff
@@ -126,53 +129,67 @@ If you want to test a branch before releasing it, you can push up a _built branc
+```
+
+#### Types
+- `feat`: new feature for the user, not a new feature for build script
````
It would be nice if this said something like "using feat will cause a minor version bump in the package once it's released" and the equivalent for all of the other commit prefixes

pretzelhammer commented on .github/CONTRIBUTING.md:
```diff
@@ -126,53 +129,67 @@ If you want to test a branch before releasing it, you can push up a _built branc
+- `feat`: new feature for the user, not a new feature for build script
+- `fix`: bug fix for the user, not a fix to a build script
+- `docs`: changes to the documentation
+- `build`: changes that affect the build system or external dependencies
```
I'm curious what happens with this prefix, because a change that affects the build system affects how the production components are built which could be a patch change, or a minor change, or major change... so how does semantic release handle this particular prefix in regards to determining how to bump the package's version?

privatenumber replied:
If deciding between `feat` or `fix` vs `build`, choose `feat` or `fix` because they influence the version bump appropriately.

If neither but it's a breaking change, use `build!`.

pretzelhammer replied:
Those are good tips it'd be nice if they were documented :)

privatenumber commented on .github/workflows/check-package-json.yml:
```diff
@@ -1,30 +0,0 @@
-    - name: Install NPM dependencies
-      run: npm ci
-    - name: Run package.json checks
-      run: node .github/dep-version-check
```
Since this just ran a single Node script, I moved it to the bottom of the PR linting workflow

privatenumber commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,24 @@
+
+on:
+  push:
+    branches: [master, next, next-major, beta, alpha]
```
Default [semantic-release branches](https://github.com/semantic-release/semantic-release/blob/5489489dda80c5ae7f047926f185ef393b131590/docs/recipes/pre-releases.md)

privatenumber commented:
Two of the workflows are marked "in-progress":
- I renamed "Lint Code & Run Tests" to "Lint", but since this is against a protected branch, it still runs the one in `master
- I deleted "Check package.json on PRs into master" and merged the script with "Lint"

privatenumber commented:
<details>
  <summary>Current semantic-release dry-run output</summary>

```
➜  maker git:(semantic-release) GH_TOKEN=<REDACTED> semantic-release -d -b semantic-release
[10:00:38 PM] [semantic-release] › ℹ  Running semantic-release version 17.4.2
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/github"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[10:00:39 PM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[10:00:47 PM] [semantic-release] › ⚠  Run automated release from branch semantic-release on repository git@github.com:square/maker.git in dry-run mode
[10:00:49 PM] [semantic-release] › ✔  Allowed to push to the Git repository
[10:00:49 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/npm"
[10:00:49 PM] [semantic-release] [@semantic-release/npm] › ℹ  Verify authentication for registry https://registry.npmjs.org/
[10:00:49 PM] [semantic-release] [@semantic-release/npm] › ℹ  Reading npm config from /Users/osame/.npmrc, /Users/osame/Documents/public-github/square/maker/.npmrc
hirokiosame
[10:00:50 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[10:00:50 PM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/github"
[10:00:50 PM] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[10:00:50 PM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/github"
[10:00:50 PM] [semantic-release] › ℹ  No git tag version found on branch semantic-release
[10:00:50 PM] [semantic-release] › ℹ  No previous release found, retrieving all commits
[10:00:50 PM] [semantic-release] › ℹ  Found 84 commits since last release
[10:00:50 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: reference semantic-release docs for pre-releases
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: added pre-release branches
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: improved conventional commits types explanation
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: github flow
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: correct step name
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm for release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm and lint
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm node version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm node version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm node version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm node version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: log node version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: log node version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: log node version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: try removing setup-node
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: try removing setup-node
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci(check-package-json): remove version check
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: automated releases with semantic release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: Add choice component (#46)

* feat: add choice component

* chore: version bump
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs(input): remove double props section (#43)
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: this actionbar is GOING TO DA MOON (#42)

* fix: this actionbar is GOING TO DA MOON

* fix: desktop

* chore: patch ver bump
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: Update button state styles (#21)

* feat: adds hover/active states to button, adjusts focus state

* fix: adjust ghost button padding

* chore: bump patch version for minor visual updates to button component
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: Select outline variant, prefix slot (#24)

* feat: adds prefix icon slot to mselect

* fix: updates mselect placeholder style

* feat: adds outline variant to mselect

* docs: add auto api docs to mselect

* fix: outline variant background for selects should be #fff

* chore: bump minor version for select component left icon slot
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: mvp Blade component (#6)

* feat: adds MVP TransitionSpringLeft component

* feat: adds MVP Blade component

* fix: broken nested blades

* docs: generate api docs for Blade and TransitionSpringLeft

* refactor: blade refactors post modal refactors

* refactor: refactor TransitionSpringLeft to use new transition system

* refactor: refactor Blade to use new transition system

* docs: improve ActionBar docs

* chore: bump minor version for new blade component
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: add border radius to loading state so rounded button doesnt have corners (#37)

* style: add border radius to loading state so button doesnt look weird

* feat: patch version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: finally fix CI
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: another ci fix
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: dead-click on ios safari for actionbar (#36)
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: refactor: transition system (breaking: v2-beta) (#35)

* refactor: makes stacked modals transition slightly better

* breaking: wip refactor of transition system

* breaking: more transition system refactor

* refactor: update transition spring up component

* breaking: refactor & rename TransitionFade to TransitionFadeIn

* refactor: transition resize component

* refactor: transition in actionbar

* refactor: nested modal transitions

* style: fix MOST lint errors

* refactor: change modal transitions again

* fix: fix transitions on resizing between mobile & tablet for modals

* chore: bump to major version pre-release

* chore: clean up commented out code

* chore: fix npm command

* chore: fix spacing & add dev deps in package.json

* chore: update package-lock.json

* chore: bump patch pre-release

* chore: bump out of pre-release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: more CI fixes
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: another minor CI fix
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: quick CI fix hopefully
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: ci check for bumping package ver
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: add install peer deps to ci
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: Publish script (#33)

* chore: wip checkout-master script

* chore: add changes

* chore: testing

* chore: testing

* chore: automate entire npm release process

* chore: lint fixes
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: Contributing docs (#29)

* docs: wip

* docs: wip

* docs: contrib docs

* docs: more contrib docs

* docs: contrib docs + update deploy index

* docs: disable unnecessary lint

* docs: fix typo

* docs: adjust title casing

* chore(deps): bump elliptic from 6.5.3 to 6.5.4 (#27)

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.5.3 to 6.5.4.
- [Release notes](https://github.com/indutny/elliptic/releases)
- [Commits](https://github.com/indutny/elliptic/compare/v6.5.3...v6.5.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* docs: fixes auto styleguide deploy script I hope

* docs: update docs

* style: lint fix

* Update .github/WHAT_IS_MAKER.md

Co-authored-by: Lokesh Dhakar <72631144+lokesh-squareup@users.noreply.github.com>

* docs: more contrib docs

* docs: more docs

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Lokesh Dhakar <72631144+lokesh-squareup@users.noreply.github.com>
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: Outline variant for MTextarea (#25)

* feat: outline variant on textarea

* docs: api tables for mtextarea

* chore: bump minor ver for outline textarea
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: adds outline variant to minput (#23)

* feat: adds outline variant to minput

* fix: removes box shadow on outline variant input focus

* fix: outline variant background for inputs should be #fff

* chore: bump minor ver
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: add styleguide deploy pr comments
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: fix lints
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: more workflow updates
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: for debugging
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: really really fix it this time
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: updates valid origins in styleguie deploy script
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: fix workflow even more
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: fix workflow?
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: refactor: styleguide deploy script updates (#31)

* refactor: styleguide deploy script updates

* refactor: test push

* refactor: add auto styleguide deploy?
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: style: update modal breakpoint (#30)

* style: update modal breakpoint

* feat: patch bump

* refactor: add tablet min width
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore(deps): bump elliptic from 6.5.3 to 6.5.4 (#27)

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.5.3 to 6.5.4.
- [Release notes](https://github.com/indutny/elliptic/releases)
- [Commits](https://github.com/indutny/elliptic/compare/v6.5.3...v6.5.4)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: remove workflow listeners on unused develop branch
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: add CODEOWNERS
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: stepper
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: update stepper to include a size slot (#28)

* feat: update stepper to include a size slot

* feat: bump version

* feat: change version bump

* style: update different sizes to all use primary button variation
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: style: update segmented control styles (#26)

* style: add slight drop shadow to selected and make unselected text color alpha 60

* feat: patch bump
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore(deps): bump pug from 3.0.0 to 3.0.2 (#20)

Bumps [pug](https://github.com/pugjs/pug) from 3.0.0 to 3.0.2.
- [Release notes](https://github.com/pugjs/pug/releases)
- [Commits](https://github.com/pugjs/pug/compare/pug@3.0.0...pug@3.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore(deps): bump pug-code-gen from 3.0.1 to 3.0.2 (#19)

Bumps [pug-code-gen](https://github.com/pugjs/pug) from 3.0.1 to 3.0.2.
- [Release notes](https://github.com/pugjs/pug/releases)
- [Commits](https://github.com/pugjs/pug/compare/pug-code-gen@3.0.1...pug-code-gen@3.0.2)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: Inline margin to padding (#22)

* fix: more predictably allocate padding for actionbar, regardless of browser quirks

* chore: bump patch version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: Auto deploy + latest deploy (#18)

* wip

* docs: sync latest versioned release to latest deploy dir

* docs: auto deploy on building lab and styleguide

* docs: adds latest styleguide link to main readme

* style: lint fixes
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: final 1.2.0 release after several 1.2.0 pre-releases
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: fix tiny bug in generate-api-tables script
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: filled input border (#17)

* fix: removes unnecessary border in input filled variant

* chore: bumps pre-release patch version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: Restacking modals (#16)

* wip

* wip

* refactor: revert stackable actionbarlayer

* docs: update modal docs after merging in restacking

* chore: bump pre-release patch version

* docs: cleans up modal stacking example

* refactor: adjust margin/padding on modalcontent & actionbarwarpper

* docs: update actionbar generated api tables

* refactor: remove responsiveactionbar

* refactor: actually remove responsive action bar
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: lint fixes
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: updates ActionBarTest
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: refactor: components inherit font-family (#14)
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: Responsive Transitions, Modals, & Actionbars (#3)

* feat: responsive modal-layer, displays modals differently on 1000px+ resolutions

* chore: minor package lock thing

* chore: update desktop breakpoint, add max width & height to modal, scroll on overflow

* refactor: mid-modal refactors, just need a savepoint

* refactor: more modal refactors

* refactor: despite this patch being a total mess I'm committing it as-is to remind myself later of the hell I had to go through to get this to work

* refactor: modal refactor & clean up

* refactor: change TransitionFade to use TransitionSpring util

* refactor: change TransitionSpringUp to use TransitionSpring

* refactor: new TransitionSpringResponsive component

* refactor: make modal use responsive transition component

* docs: updates docs post-refactor

* refactor: move pan down behavior to modal from modalcontainer

* feat: mvp transition resize component

* docs: mighty morphin modals example

* refactor: make TransitionResize animation faster

* refactor: update transitionresize speeds

* feat: disable pan down close on modal prop

* docs: update modal readme

* feat: actionbar slot in modals

* docs: generate modal docs

* docs: styleguide-specific actionbarlayer hacks

* fix: minor ActionBar fixes

* docs: update modal stacking example to use action bars

* refactor: inline actionbar component

* refactor: move modal actionbar slot into its own component

* docs: clean up modal docs

* refactor: removes ModalContainer component

* refactor: get rid of stacking garbage from modal layer

* refactor: remove nebulous pan down functionality

* docs: modal docs

* docs: adds keyed transitions example to actionbar docs

* refactor: removes tertiary variant from actionbarbutton

* chore: adds refactor TODO comment

* docs: update modal docs example to make it more flow-y

* refactor: removes unused variants on ActionBarButtons

* chore: bump minor version pre-release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: patch update for segcon design bug fixes
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: segcon sizing (#13)

* fix: segcon sizing

* fix: segcon medium & small variant font size & line height
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Torch subslot (#12)

* fix: torch subslot, was breaking msegment in released builds

* chore: bump patch version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: patch version bump
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Disable segmentedcontrol validation (#11)

* fix: disable subslot validation for SegmentedControl

* chore: bump patch version
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: segmented control
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: feat: mvp SegmentedControl component (#5)

* feat: mvp SegmentedControl component

* feat: mvp SegmentedControl component

* feat: adds shape & size props to SegmentedControl

* chore: increment lib version number to signal addition of new component
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is minor
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: wip site app lab demo (#10)

* docs: wip app site lab demo

* docs: rename some stuff

* docs: silence annoying lint warnings
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: comment out problematic styles (#8)
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: Lab github pages (#7)

* docs: deploy lab and styleguide built assets into same dist dir, and generate index file for deploys

* docs: generate index after lab or styleguide deploys

* docs: minor touch-up

* docs: integration branches for styleguide deploys
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: ignore dist dir
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: adds npmrc
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: make docs homepage same as github repo readme (#4)
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: removes orbit-icons, uses better maker-icons (#1)

* fix: removes orbit-icons, uses better maker-icons

* chore: fix icons some more

* chore: adds menu icon back to lbaoratory component

* chore: bumps maker-icons

* chore: bump version from alpha to v1

* docs: minor readme update
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: fix: weird hover styles in KitchenSink lab (#2)
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The release type for the commit is patch
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: bumps version number
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: chore: updates package.json license info
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: Make it so.

Co-authored-by: Hiroki Osame <hiroki.osame@gmail.com>
Co-authored-by: pretzelhammer <7769424+pretzelhammer@users.noreply.github.com>
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[10:00:50 PM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 84 commits complete: minor release
[10:00:50 PM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[10:00:50 PM] [semantic-release] › ℹ  There is no previous release, the next release version is 1.0.0
[10:00:50 PM] [semantic-release] › ℹ  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[10:00:51 PM] [semantic-release] › ✔  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[10:00:51 PM] [semantic-release] › ⚠  Skip step "prepare" of plugin "@semantic-release/npm" in dry-run mode
[10:00:51 PM] [semantic-release] › ⚠  Skip v1.0.0 tag creation in dry-run mode
[10:00:51 PM] [semantic-release] › ⚠  Skip step "publish" of plugin "@semantic-release/npm" in dry-run mode
[10:00:51 PM] [semantic-release] › ⚠  Skip step "publish" of plugin "@semantic-release/github" in dry-run mode
[10:00:51 PM] [semantic-release] › ⚠  Skip step "success" of plugin "@semantic-release/github" in dry-run mode
[10:00:51 PM] [semantic-release] › ✔  Published release 1.0.0 on default channel
[10:00:51 PM] [semantic-release] › ℹ  Release note for version 1.0.0:
# 1.0.0 (2021-04-01)

### Bug Fixes

    * add border radius to loading state so rounded button doesnt have corners (#37) (0be651a)
    * dead-click on ios safari for actionbar (#36) (ec7a64d)
    * filled input border (#17) (62181b6)
    * Inline margin to padding (#22) (2693c8c)
    * removes orbit-icons, uses better maker-icons (#1) (a5fbe0f)
    * segcon sizing (#13) (41baffa)
    * this actionbar is GOING TO DA MOON (#42) (6731eee)
    * weird hover styles in KitchenSink lab (#2) (acd3865)

### Features

    * Add choice component (#46) (cb5dc65)
    * adds outline variant to minput (#23) (6ea6a1e)
    * mvp Blade component (#6) (8ee55e2)
    * mvp SegmentedControl component (#5) (7f178c8)
    * Outline variant for MTextarea (#25) (c8fe160)
    * Responsive Transitions, Modals, & Actionbars (#3) (77f3ae7)
    * Restacking modals (#16) (3278a39)
    * Select outline variant, prefix slot (#24) (02bbea8)
    * Update button state styles (#21) (87028e8)
    * update stepper to include a size slot (#28) (96dca07)
```

</details>


Note, it's not detecting the latest version because the releases didn't have git tags published. I will back fill them so semantic-release can detect the right version.

privatenumber commented:
Updated the Git tags with each versioned release. Output looking good now:

<details>
  <summary>Current semantic-release dry-run output</summary>

```
➜  maker git:(semantic-release) GH_TOKEN=<REDACTED> semantic-release -d -b semantic-release
[3:26:10 AM] [semantic-release] › ℹ  Running semantic-release version 17.4.2
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "verifyConditions" from "@semantic-release/github"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "prepare" from "@semantic-release/npm"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/npm"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "publish" from "@semantic-release/github"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/npm"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "addChannel" from "@semantic-release/github"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "success" from "@semantic-release/github"
[3:26:10 AM] [semantic-release] › ✔  Loaded plugin "fail" from "@semantic-release/github"
[3:26:13 AM] [semantic-release] › ⚠  Run automated release from branch semantic-release on repository git@github.com:square/maker.git in dry-run mode
[3:26:14 AM] [semantic-release] › ✔  Allowed to push to the Git repository
[3:26:14 AM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/npm"
[3:26:14 AM] [semantic-release] [@semantic-release/npm] › ℹ  Verify authentication for registry https://registry.npmjs.org/
[3:26:14 AM] [semantic-release] [@semantic-release/npm] › ℹ  Reading npm config from /Users/osame/.npmrc, /Users/osame/Documents/public-github/square/maker/.npmrc
hirokiosame
[3:26:16 AM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[3:26:16 AM] [semantic-release] › ℹ  Start step "verifyConditions" of plugin "@semantic-release/github"
[3:26:16 AM] [semantic-release] [@semantic-release/github] › ℹ  Verify GitHub authentication
[3:26:16 AM] [semantic-release] › ✔  Completed step "verifyConditions" of plugin "@semantic-release/github"
[3:26:16 AM] [semantic-release] › ℹ  Found git tag v2.2.3 associated with version 2.2.3 on branch semantic-release
[3:26:16 AM] [semantic-release] › ℹ  Found 18 commits since last release
[3:26:16 AM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: reference semantic-release docs for pre-releases
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: added pre-release branches
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: improved conventional commits types explanation
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: docs: github flow
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: correct step name
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm for release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm and lint
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm node version
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm node version
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm node version
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: use nvm node version
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: log node version
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: log node version
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: log node version
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: try removing setup-node
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: try removing setup-node
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci(check-package-json): remove version check
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analyzing commit: ci: automated releases with semantic release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  The commit should not trigger a release
[3:26:16 AM] [semantic-release] [@semantic-release/commit-analyzer] › ℹ  Analysis of 18 commits complete: no release
[3:26:16 AM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[3:26:16 AM] [semantic-release] › ℹ  There are no relevant changes, so no new version is released.
```

</details>

pretzelhammer commented on .github/CONTRIBUTING.md:
````diff
@@ -127,52 +130,70 @@ If you want to test a branch before releasing it, you can push up a _built branc
-```
+
+#### Pre-releases
+For instructions on how to do pre-releases, refer to the [semantic-release docs](https://github.com/semantic-release/semantic-release/blob/master/docs/recipes/pre-releases.md).
````
nice, thanks for adding this

pretzelhammer commented:
@privatenumber this looks good to me! I'm down for us to start trying it out for real.
